core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -3643,6 +3649,34 @@ int lookupPositionForOffset(RemoteLogSegmentMetadata remoteLogSegmentMetadata, l
         assertEquals(273, fetchDataInfo.fetchOffsetMetadata.relativePositionInSegment);
     }
 
+    @Test
+    public void testRLMOpsWhenMetadataIsNotReady() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(2);
+        when(remoteLogMetadataManager.isReady(any(TopicIdPartition.class)))
+                .thenAnswer(ans -> {
+                    latch.countDown();
+                    return false;
+                });
+        remoteLogManager.startup();
+        remoteLogManager.onLeadershipChange(
+                Collections.singleton(mockPartition(leaderTopicIdPartition)),
+                Collections.singleton(mockPartition(followerTopicIdPartition)),
+                topicIds
+        );
+        assertNotNull(remoteLogManager.rlmCopyTask(leaderTopicIdPartition));
+        assertNotNull(remoteLogManager.leaderExpirationTask(leaderTopicIdPartition));
+        assertNotNull(remoteLogManager.followerTask(followerTopicIdPartition));
+
+        latch.await(5, TimeUnit.SECONDS);
```

Comment:
The earlier `remoteLogManager.onLeadershipChange` should have called `remoteLogMetadataManager.isReady` and return true. But if there are any regressions, it will wait for 5 seconds and the test will fail. Feel free to add a similar comment in the code.